app/controllers/contacts_controller.rb

```diff
@@ -0,0 +1,21 @@
+class ContactsController < ApplicationController
+  def new  
+    @company = Company.find(params[:id])
+  end
+
+  def create
+    @contact = Contact.new(contact_params)
+    @contact.company_id = params[:company_id]
+    if @contact.save
+      flash[:success] = "Created Contact!"
+      redirect_to company_jobs_path(@contact.company)
+    else
+      redirect_to company_jobs_path(@contact.company)
```

Comment:
again, be sure you handle the errors - no need to redirect to the same place in an if statement. if you're going to be redirecting to the same place, then do this after the conditional